### ISSUE TITLE
Overwritten request headers (testcafe@2.6.0)

### ISSUE THREAD

**Issue by gabrielrobert** [DevExpress/testcafe#7732]
### What is your Scenario?

I'm utilizing TestCafé to log in to a platform and carry out certain tasks. It functions seamlessly until we make the transition from testcafe@2.5.0 to testcafe@2.6.0.

### What is the Current behavior?

The server receives the request, but the headers "Accept" and "Content-Type" get overwritten, leading to a 415 error.

Accept: */*
Content-Type: text/plain;charset=UTF-8

![2023-05-26 15_44_40-DevTools - localhost_3000__redirect=login returnTo=http%3a%2f%2flocalhost%3a2000](https://github.com/DevExpress/testcafe/assets/5783827/7538a670-a6db-4220-89cd-b4c889eefc0f)

### What is the Expected behavior?

The headers "Accept" and "Content-Type" remain intact. Here's an example of using testcafe@2.5.0 without any modifications to the code.

Accept: application/json, text/plain, */*
Content-Type: application/json

![2023-05-26 15_55_21-DevTools - localhost_2000_seed-event-with-virtual_participants](https://github.com/DevExpress/testcafe/assets/5783827/4c9b073a-0f10-454c-a9a0-e09ff44e4dfb)

### What is your public website URL? (or attach your complete example)

https://dashboard.fourwaves.com/login

### What is your TestCafe test code?

```
 public async Login() {
    return await t
      .navigateTo('http://localhost:3000/login')
      .typeText(Selector('[name="Email"]'), 'random_email@test.com')
      .typeText(Selector('[name="Password"]'), '****')
      .click(Selector('.js-submit'))
  }
```

### Your complete configuration file

```
module.exports = {
  quarantineMode: false,
  skipJsErrors: true,
  selectorTimeout: 15000,
  assertionTimeout: 10000,
  concurrency: 1,
  screenshots: {
    path: 'tests/e2e/!screenshots',
    takeOnFails: true,
    fullPage: true,
    thumbnails: false,
  },
  runExecutionTimeout: 600000,
  testExecutionTimeout: 600000,
  pageRequestTimeout: 600000,
  nativeAutomation: true,
  hooks: {
    test: {
      before: async t => {
        await t.setCookies({ i18n_redirected: 'en' }, 'http://localhost:3000');
      },
    },
  },
};
```

### Your complete test report

Unfortunately, the test report is irrelevant as it suggests that the login was successful, but it is unable to locate the next selector.

### Steps to Reproduce

1. Perform an upgrade to testcafe@2.6.0.
2. Execute the tests.
3. Observe errors from the test run.
4. Revert back to testcafe@2.5.0.
5. Observe the successful tests.

### TestCafe version

2.6.0

### Node.js version

v16.19.0

### Command-line arguments

testcafe chrome tests/e2e

### Browser name(s) and version(s)

Chrome 113

### Platform(s) and version(s)

Windows

**Comment by miherlosev** [comment]
Hi @gabrielrobert,
Could you please modify the example so that it's possible to run it locally without any additional setup (see [How to create a minimal working example](https://testcafe.io/documentation/402636/faq/general-info?search#how-to-create-a-minimal-working-example-when-you-submit-an-issue) article)?

**Comment by gabrielrobert** [comment]
Fixed by testcafe@2.6.2.
